schemaregistry/schemaregistry.md

```diff
@@ -0,0 +1,257 @@
+concurrently. That implies that the schema managed by this registry is not
+a single document, but MAY be a set of documents that all describe the exact
+same data structure, each in their own particular syntax and using their own 
+type model.
+
+The schema version therefore relates to this abstract notion of schema and the
+described data structure. All documents coexisting within the same version
+SHOULD describe the exact same data structure.
+
+### 2.2.2. Schema attributes
+
+As per the definition above, the schema object is a collection of versions
+of schema documents. It only has an identifier attribute.
+
+#### id
+
+- Type: `String`
+- Description: Identifies the schema.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string
+  - MUST conform with RFC3986/3.3 `segment-nz-nc` syntax
+  - MUST be unique within the scope of the schema registry
+- Examples:
+  - myschema
+  - my-schema
+
+### 2.2.3 Schema version attributes
+
+A schema version is a document. The "body" of a schema version MAY be a text
+document or binary stream. An implementation SHOULD validate whether a
+schema version is valid according to the rules of its format, for instance
+whether it is a valid Avro schema document when the format is Apache Avro.
+
+Within the scope of the schema set, the version is identified by the combination
+of a version number and an optional format identifier. The schema version MAY
+also have an additional, optional unique identifier within the scope of the
+registry.
+
+#### version
+
+- Type: `Integer`
+- Description: The version of the schema. This is a simple counter and tracks
+  the version in the scope of this schema within the schema group. The schema
+  document MAY indicate a schema that follows a different versioning scheme.
+- Constraints:
+  - REQUIRED
+  - Assigned by server.
+- Examples:
+  - 1
+  - 2
+
+#### format
+
+- Type: `String`
+- Description: 
+- Constraints:
+  - OPTIONAL. Can be used if and only if not format has been set for the schema
+    group hosting the schema.
+  - MUST be a non-empty string, if present.
+- Examples:
+  - "avro"
+  - "json-schema"
+  - "xsd11"
+
+#### id
+
+- Type: `String`
+- Description: Identifies the schema document uniquely without requiring
+  other qualifiers.
+- Constraints:
+  - OPTIONAL
+  - Assigned by the server.
+- Examples:
+  - { ... guid ... }  
+
+## 3. HTTP ("REST") API 
+
+This section informally describes the HTTP API binding of this schema registry.
+The formal definition is the [OpenAPI document](./schemaregistry.yaml) that is
+part of this specification.
+
+### 3.1. Path hierarchy
+
+Schema groups contain schemas and those contain schema versions, which are the
+documents required for serialization or validation.
+
+These dependencies are reflected in the path structure:
+
+`/schemagroups/{group-id}/schemas/{schema-id}/versions/{version}`
+
+The name of the first segment of the path is a suggestion and MAY differ between
```

Comment:
How do you see the query for all group getting interop w/o agreement on the first one?

Reply:
"schemagroups" is really part of the path to the registry itself; everything defined here sits under that path. the segment could even be empty or could be multiple segments. I don't see there being interop issues, because you would always have it being at the root of the URI.

Reply:
ah, I didn't realize you meant for "schemagroups" to be an impl choice - we might want to make clearer throughout then entire doc

Reply:
Also update the OpenAPI to remove `/schemagroups` from paths.  Would make it easier to generate servers/clients from the OpenAPI at some future point.